eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/RecordId.scala

```diff
@@ -14,21 +17,25 @@ import com.evolutiongaming.scassandra.*
   * that deleted record and recreated one are not considered the same while sharing
   * same private key.
   */
-final case class RecordId(value: String) extends AnyVal {
-  override def toString: String = value
+final case class RecordId(value: UUID) extends AnyVal {
+  override def toString: String = value.toString()
```

Comment:
in other places, where we want to disable default `apply` (to force usage of `of` or `unsafe`), we use `sealed abstract case class` instead of regular case class

Reply:
I think you used false assumption here: I do not see default `apply` as dangerous constructor and do not want disabling it. Mainly because creating new `RecordId` via default `apply` require pre-generated `UUID`, so unsafe action should be done _before_ `RecordId` creation.

In `SeqNr` Yaroslav disabled constructor because sequence numbers are applicable for arithmetic operations, which are dangerous. `RecordId`s have only one relation between each other - equality, thus protecting against mutation of `UUID` is pointless (each mutation gives unequal to original `UUID` while `SeqNr` arithmetics not so strait-forward).

My reasoning mentioned before still are not enough to keep the `apply` public, because its actually _not required_ to be public 😄 . Actual thing that stops me from following your suggestion is impossibility of making _abstract value class_: value class (one that extends `AnyVal`) is final (even if its not marked as `final`) thus it cannot be `abstract` - that's it.

Alternatively I can mark the constructor as private:
```
final case class RecordId private (value: UUID) extends AnyVal
``` 
but that makes Scala 3 compiler unhappy:
```
[error] <dev dir>/kafka-journal/eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/RecordId.scala:20:18: access modifiers for `copy` method are copied from the case class constructor under Scala 3 (or with -Xsource-features:case-apply-copy-access)
[error] Scala 3 migration messages are issued as errors under -Xsource:3. Use -Wconf or @nowarn to demote them to warnings or suppress.
[error] Applicable -Wconf / @nowarn filters for this fatal warning: msg=<part of the message>, cat=scala3-migration, site=com.evolutiongaming.kafka.journal.eventual.cassandra.RecordId
[error] final case class RecordId private (value: UUID) extends AnyVal {
[error]                  ^
[error] one error found
``` 

I prefer keeping the `apply` publicly available, as its now.